### PR TITLE
Adds 3 second cooldown to containment fields generating sparks

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
They can still throw you/murder you forever but it will only make sparks every three seconds because you can basically crash the server with this otherwise.

### CODE REVIEW

File: code/modules/power/singularity/containment_field.dm [hunk 1]
```diff
@@ -66,6 +66,7 @@
 
 /obj/machinery/field
 	var/hasShocked = 0 //Used to add a delay between shocks. In some cases this used to crash servers by spawning hundreds of sparks every second.
```
Comment (Anonus) [inline code/modules/power/singularity/containment_field.dm]: was this var even used

**Comment by Aranclanos** [comment]
just put a fucking spawn you retarded monkey
oh wait there's already one in the line 112 in a var already made to do exactly this
the var that is one line on top of the new one that you made

**Comment by KorPhaeron** [comment]
Wouldn't spawn() still fuck up if you threw multiple items at the field so it was constantly resetting the var?

File: code/modules/power/singularity/containment_field.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/power/singularity/containment_field.dm]: Because it isn't taken into account at all when you throw objects at the field, only mobs

**Comment by KorPhaeron** [comment]
What is wrong with the world.time thing?

**Comment by RemieRichards** [comment]
>Mr why the fuck did you use ```spawn()``` advocating using ```spawn()```

Ok w/e.

**Comment by KorPhaeron** [comment]
@RemieRichards 

That's why I'm confused

**Comment by optimumtact** [comment]
NO

use mso's timer subsystem, have it schedule a new run of itself every 3 seconds

**Comment by WJohn** [comment]
Or wait this isn't a balance PR, more a fix than anything.

**Comment by Aranclanos** [comment]
not listening etc

**Comment by Shapsy** [comment]
Uhh leaving this open leaves the server vulnerable to people crashing it

**Comment by Anonus** [comment]
>insisting on using spawn

**Comment by KorPhaeron** [comment]
I asked you to explain why, given you usually say not to use spawn, how is that not listening?

Attempting to understand something isnt a failure to listen

**Comment by optimumtact** [comment]
@Aranclanos if you can't communicate effectively don't expect people to implement what you're asking for.

perhaps instead of throwing your toys out of the cot, you could try to explain what you mean in a way that normal human beings can understand.

**Pull request closed without merging**